src/platform/linux/wayland/window.rs

```diff
@@ -235,6 +242,8 @@ impl Window {
         } else {
             self.frame.lock().unwrap().unset_fullscreen();
         }
+
+        *(self.fullscreen.lock().unwrap()) = monitor.clone();
```

Comment:
Actually, this would not be the proper place to fill this value. Wayland is all about requesting permission, and we are not sure that we will be set fullscreen on the monitor we requested (or set fullscreen at all actually). We need to track it from the `Configure` events we get.

At this line:

https://github.com/rust-windowing/winit/blob/b94572621a4da21af909950d8a00211b053c4d5f/src/platform/linux/wayland/window.rs#L47-L49

The [`Configure` variant](https://docs.rs/smithay-client-toolkit/0.6.1/smithay_client_toolkit/window/enum.Event.html#variant.Configure) has an other field, `states`. Which is a `Vec`. If we are set fullscreen, it'll contain [`State::Fullscreen`](https://docs.rs/smithay-client-toolkit/0.6.1/smithay_client_toolkit/window/enum.State.html#variant.Fullscreen) as one of its values.

You can then obtain the monitor we are fullscren on using `get_outputs(&my_surface)` in a way similar to the `get_current_monitor` method:

https://github.com/rust-windowing/winit/blob/b94572621a4da21af909950d8a00211b053c4d5f/src/platform/linux/wayland/window.rs#L273-L279

Reply:
After trying to implement this myself, I'm not sure that it's possible with the current architecture.

My original plan was to check for `State::Fullscreen`, toggle a `fullscreen` `bool` and then make `get_fullscreen()` something like:

```rs
pub fn get_fullscreen(&self) -> Option<MonitorId> {
  if self.fullscreen {
    Some(self.get_current_monitor())
  else {
    None
  }
}

However, the closure that's checking the for the `event::Configure` variant only has access to the `InternalWindow` structs. However, the `get_fullscreen()` method only has access to the `Window` struct.

I don't know how to share any information between these two scopes with the current implementation, so I'm not sure where to go from here.

@vberger any chance you could shed some light on a potential solution?

Reply:
You can share the `fullscreen` boolean between the `Window` and `InternalWindow`, in the same way their `size` field is shared.

Reply:
Ah! Of course, I had missed that. I'll do that now. :)